utils/python/CIME/XML/compilerblock.py

```diff
@@ -115,6 +117,35 @@ def _handle_references(self, elem, set_up, tear_down, depends):
         output = elem.text
         if output is None:
             output = ""
+
+        reference_re = re.compile(r'\${?(\w+)}?')
```

Comment:
Because it doesn't handle it in quite the same way - 
In generic $ENV{NETCDF} is resolved at run time.   In compilerblock it is converted to make or cmake syntax but not resolved until used.